src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -691,4 +692,29 @@ static boolean convertStringToBoolean(String value) {
         || "true".equals(lowerCasedValue)
         || "on".equals(lowerCasedValue);
   }
+
+  /**
+   * Create exception when value cannot be ingested into column of a specific type.
+   *
+   * @param value Invalid value causing the exception
+   * @param type Snowflake column type
+   */
+  private static SFException cannotConvertException(Object value, String type) {
+    return new SFException(
+        ErrorCode.INVALID_ROW,
+        sanitizeValueForExceptionMessage(value),
+        String.format("Value cannot be converted to %s", type));
+  }
+
+  /**
+   * Before passing a value to an exception string, we should limit how many characters are
+   * displayed. Important especially for "value exceeds max column size" exceptions.
+   *
+   * @param value Value to adapt for exception message
+   */
+  private static String sanitizeValueForExceptionMessage(Object value) {
```

Comment:
I'm not sure if this is really needed, what if customer need to copy the value?

Reply:
You mean copy from the logs? We can increase the limit, but I think there should be some limit. Previously if somebody tried to ingest a 100MB string, SDK rejects it, but dump those 100MB into the logs.